src/core/service.c

```diff
@@ -2437,6 +2602,10 @@ static int service_deserialize_item(Unit *u, const char *key, const char *value,
                         s->watchdog_override_enable = true;
                         s->watchdog_override_usec = watchdog_override_usec;
                 }
+        } else if (STR_IN_SET(key, "main-command", "control-command")) {
+                r = service_deserialize_exec_command(u, key, value);
+                if (r < 0)
+                        log_unit_debug(u, "Failed to parse previously executed command: %s", value);
```

Comment:
`log_unit_debug_errno(u, r, "Failed to parse serialized command \"%s\": %m", value);`